internal/cortex/querier/queryrange/query_range.go

```diff
@@ -217,9 +227,25 @@ func (prometheusCodec) MergeResponse(_ Request, responses ...Response) (Response
 	// we need to pass on all the headers for results cache gen numbers.
 	var resultsCacheGenNumberHeaderValues []string
 
+	var headers []*PrometheusResponseHeader
+	// merge headers
 	for _, res := range responses {
 		promResponses = append(promResponses, res.(*PrometheusResponse))
 		resultsCacheGenNumberHeaderValues = append(resultsCacheGenNumberHeaderValues, getHeaderValuesWithName(res, ResultsCacheGenNumberHeaderName)...)
```

Comment:
I doubt this matters given what was happening before, but this could be out of sync with the actual headers we return in the response due to the merging below.